src/core/ext/filters/http/client/http_client_filter.cc

```diff
@@ -125,17 +129,23 @@ ArenaPromise<ServerMetadataHandle> HttpClientFilter::MakeCallPromise(
 }
 
 HttpClientFilter::HttpClientFilter(HttpSchemeMetadata::ValueType scheme,
-                                   Slice user_agent)
-    : scheme_(scheme), user_agent_(std::move(user_agent)) {}
+                                   Slice user_agent,
+                                   bool test_only_use_put_requests)
+    : scheme_(scheme),
+      user_agent_(std::move(user_agent)),
+      test_only_use_put_requests_(test_only_use_put_requests) {}
 
 absl::StatusOr<HttpClientFilter> HttpClientFilter::Create(ChannelArgs args,
                                                           ChannelFilter::Args) {
   auto* transport = args.GetObject<grpc_transport>();
   if (transport == nullptr) {
     return absl::InvalidArgumentError("HttpClientFilter needs a transport");
   }
-  return HttpClientFilter(SchemeFromArgs(args),
-                          UserAgentFromArgs(args, transport->vtable->name));
+  absl::optional<int> use_put_requests =
```

Comment:
This can be:

```
bool use_put_requests =
    args.GetInt(GRPC_ARG_TEST_ONLY_USE_PUT_REQUESTS).value_or(false);
```

You can probably even just avoid the local variable and inline this expression in the `HttpClientFilter` instantiation below.

Reply:
Done!